query/src/main/java/com/connexta/replication/query/QueryService.java

```diff
@@ -0,0 +1,219 @@
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package com.connexta.replication.query;
+
+import com.connexta.replication.api.AdapterException;
+import com.connexta.replication.api.NodeAdapter;
+import com.connexta.replication.api.data.Filter;
+import com.connexta.replication.api.data.FilterIndex;
+import com.connexta.replication.api.data.Metadata;
+import com.connexta.replication.api.data.MetadataInfo;
+import com.connexta.replication.api.data.OperationType;
+import com.connexta.replication.api.data.QueryRequest;
+import com.connexta.replication.api.data.Site;
+import com.connexta.replication.api.data.TaskInfo;
+import com.connexta.replication.api.queue.QueueException;
+import com.connexta.replication.api.queue.SiteQueue;
+import com.connexta.replication.data.QueryRequestImpl;
+import com.google.common.annotations.VisibleForTesting;
+import java.time.Instant;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The <code>QueryService</code> class implements the logic for querying a specific DDF-based site
+ * for changes to be replicated or harvested.
+ */
+public class QueryService {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(QueryService.class);
+
+  private final QueryServiceTools queryServiceTools;
+
+  private final ScheduledExecutorService executor;
+
+  /** The current configuration for the site being queried by this query service. */
+  private Site site;
+
+  private SiteQueue siteQueue;
+
+  /** The adapter to communicate with the given site. */
+  private NodeAdapter adapter;
+
+  @VisibleForTesting ScheduledFuture future;
+
+  /**
+   * Instantiates a new query service.
+   *
+   * @param site the configuration for the site to be queried by this query service
+   */
+  public QueryService(Site site, QueryServiceTools queryServiceTools) {
+    this(site, queryServiceTools, Executors.newSingleThreadScheduledExecutor());
+  }
+
+  public QueryService(
+      Site site, QueryServiceTools queryServiceTools, ScheduledExecutorService executor) {
+    this.queryServiceTools = queryServiceTools;
+    this.siteQueue = queryServiceTools.getQueueFor(site);
+    this.executor = executor;
+    update(site);
```

Comment:
❗️ Should not be calling public methods from a constructor unless the class is final

Reply:
I guess I'll just make the class final. Or I think I could just make the method final.

Reply:
Nvm I just learned that you can't mock/spy a final class and since we are going to let the adapter exception bubble out I think I'll just set the fields directly.